tool_path_planner/src/raster_tool_path_planner.cpp

```diff
@@ -30,19 +29,90 @@
 #include <vtkTriangleFilter.h>
 
 #include <pcl/surface/vtk_smoothing/vtk_utils.h>
-
-//#include <pcl/pcl_tests.h>
-
 #include <pcl/point_types.h>
 #include <pcl/sample_consensus/ransac.h>
 #include <pcl/sample_consensus/sac_model_plane.h>
+
+#include <log4cxx/basicconfigurator.h>
+#include <log4cxx/patternlayout.h>
+#include <log4cxx/consoleappender.h>
 #include <tool_path_planner/raster_tool_path_planner.h>
+#include "tool_path_planner/utilities.h"
+
+
+static const std::size_t MAX_ATTEMPTS = 1000;
+static const double ANGLE_CORRECTION_THRESHOLD = (150.0/180.0)*M_PI;
+static const double EXTRUDE_EXTEND_PERCENTAGE = 1.5;
+static const double RAY_INTERSECTION_TOLERANCE = 0.001;
+
+
+log4cxx::LoggerPtr createConsoleLogger(const std::string& logger_name)
+{
+  using namespace log4cxx;
+  PatternLayoutPtr pattern_layout(new PatternLayout( "[\%-5p] [\%c](L:\%L): \%m\%n"));
+  ConsoleAppenderPtr console_appender(new ConsoleAppender(pattern_layout));
+  log4cxx::LoggerPtr logger(Logger::getLogger(logger_name));
+  logger->addAppender(console_appender);
+  logger->setLevel(Level::getInfo());
+  return logger;
+}
+
+/**
+ * @brief computes the angle between two vectors
+ * @param v1
+ * @param v2
+ * @return The angle in radians
+ */
+double computeAngle(const Eigen::Vector3d& v1, const Eigen::Vector3d& v2)
+{
+  return std::acos(v1.dot(v2)/(v1.norm() * v2.norm()));
+}
+
+/**
+ * @brief computes the squared distance between two 3d vectors
+ * @param pt1
+ * @param pt2
+ * @return
+ */
+double computeSquaredDistance(const std::vector<double>& pt1, const std::vector<double>& pt2)
+{
+  if(pt1.size() != 3 || pt2.size() != 3)
+  {
+    return 0;
+  }
+  return (pow(pt1[0] - pt2[0], 2.0) + pow(pt1[1] - pt2[1], 2.0 ) + pow((pt1[2] - pt2[2]), 2.0 ));
+}
+
+/**
+ * @brief findClosestPoint Finds the closest point in a list to a target point
+ * @param pt The target point
+ * @param pts The list of points to search for the closest point
+ * @return The index of the closest point
+ */
+int findClosestPoint(const std::vector<double>& pt,  const std::vector<std::vector<double> >& pts)
+{
+  double min = std::numeric_limits<double>::max();
+  int index = -1;
+  for(int i = 0; i < pts.size(); ++i)
+  {
+    double d = computeSquaredDistance(pt, pts[i]);
+    if(d < min)
+    {
+      index = i;
+      min = d;
+    }
+  }
+  return index;
+}
+
+
+log4cxx::LoggerPtr tool_path_planner::RasterToolPathPlanner::RASTER_PATH_PLANNER_LOGGER = createConsoleLogger("RasterPathPlanner");
+
 
 namespace tool_path_planner
 {
 
-  RasterToolPathPlanner::RasterToolPathPlanner(bool use_ransac):
-      use_ransac_normal_estimation_(use_ransac)
+  RasterToolPathPlanner::RasterToolPathPlanner()
   {
     debug_on_ = false;
```

Comment:
Rather than declaring a variable that must be set to true for debug to run, perhaps you could use a 
#define TOOL_PATH_PLANNER_DEBUG_ON
and then use
#if(TOOL_PATH_PLANNER_DEBUG_ON),
The way its written now with yield a compiler warning about unreachable code.